### PR TITLE
overlays: Update dependencies and amend the script to add headers.

### PR DESCRIPTION
**Pull request by erg**

### CODE REVIEW

File: stdlib/public/SDK/os/CMakeLists.txt [hunk 1]
```diff
@@ -10,11 +10,11 @@ add_swift_library(swiftos ${SWIFT_SDK_OVERLAY_LIBRARY_BUILD_TYPES} IS_SDK_OVERLA
...
   SWIFT_MODULE_DEPENDS_OSX Darwin Dispatch ObjectiveC XPC # auto-updated
-  SWIFT_MODULE_DEPENDS_IOS Darwin ObjectiveC # auto-updated
+  SWIFT_MODULE_DEPENDS_IOS Darwin Dispatch ObjectiveC XPC # auto-updated
     Dispatch # required in some configurations but not found by tool
-  SWIFT_MODULE_DEPENDS_TVOS Darwin ObjectiveC # auto-updated
+  SWIFT_MODULE_DEPENDS_TVOS Darwin Dispatch ObjectiveC XPC # auto-updated
     Dispatch # required in some configurations but not found by tool
-  SWIFT_MODULE_DEPENDS_WATCHOS Darwin ObjectiveC # auto-updated
+  SWIFT_MODULE_DEPENDS_WATCHOS Darwin Dispatch ObjectiveC XPC # auto-updated
     Dispatch # required in some configurations but not found by tool
```
Comment (jrose-apple) [inline stdlib/public/SDK/os/CMakeLists.txt]: Redundant, although it doesn't matter.

File: utils/find-overlay-dependencies.sh [hunk 2]
```diff
@@ -64,9 +64,10 @@ CMAKE_DEPENDS_NAME[watchos]="SWIFT_MODULE_DEPENDS_WATCHOS"
 
 echo $1
 for sdk in ${(k)SDKS}; do
+  sdk=$sdk$SUFFIX
   arch=$SDKS[$sdk]
```
Comment (jrose-apple) [inline utils/find-overlay-dependencies.sh]: Don't we still need the original name for looking up into SDKS and CMAKE_DEPENDS_NAME?

Comment (erg) [inline utils/find-overlay-dependencies.sh]: Yes, fixing that.

**Comment by erg** [comment]
@swift-ci Please smoke test and merge

**Comment by jrose-apple** [comment]
Whoops, right, things can't depend on XPC on iOS because XPC isn't available on iOS.

**Pull request merged**